### DISCUSSION TITLE
Cannot connect  three harvester to the main farmer node

### DISCUSSION THREAD

**Discussion by LTMI** [Chia-Network/chia-blockchain#17250] [Support]
Hi support,

I have 4 nodes (each node has a 16TB SATA hard drive) . 1 main farmer that connect to three harvester. It used to work until I have moved these nodes to other  
The farmer node will be able to ping to the other three nodes. but the main farmer node show Total Harvester is 1 instead of 4. Please kindly advise what should I do next in order to connect the other three harvesters to the farmer node. I am using Windows 10 Pro and current Chia version is 1.5 . Thank you for your help.

**Comment by BrandtH22** [comment]
Hey @LTMI  , as a heads up we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia).

What I would do is upgrade all machines to the latest and have new configs created:
1. upgrade to the latest release candidate (there are peer related optimizations): https://github.com/Chia-Network/chia-blockchain/releases/tag/2.1.4-rc2

2. While installing that I would remove the current peers.dat file in ~\.chia\mainnet\db\peers.dat (this will force the client to connect to new peers upon restarting)

3. Once the latest RC is installed, have a new config file created:
 - Stop chia
 - Rename the config file at .chia\mainnet\config\config.yaml
 - Run chia init
 - Copy any custom settings from the old config to the new one (plot directories, pool information, database path, trusted nodes etc) - there are more automated ways to migrate that info if not comfortable editing the config file, yaml is very picky about formatting.
 - Follow the setup for farming on multiple machines here https://docs.chia.net/farming-on-many-machines/
 - Start chia and let it sync

Let us know if you run into any issues or have questions during the process.

**Comment by LTMI** [comment]
Thank you so much for your quick reply. I will go ahead upgrade to the latest version like you recommended. On Jan 9, 2024, at 10:59 AM, ClydeWallace22 ***@***.***> wrote:﻿
Hey @LTMI  , as a heads up we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia).
What I would do is upgrade all machines to the latest and have new configs created:


upgrade to the latest release candidate (there are peer related optimizations): https://github.com/Chia-Network/chia-blockchain/releases/tag/2.1.4-rc2


While installing that I would remove the current peers.dat file in ~.chia\mainnet\db\peers.dat (this will force the client to connect to new peers upon restarting)


Once the latest RC is installed, have a new config file created:



Stop chia
Rename the config file at .chia\mainnet\config\config.yaml
Run chia init
Copy any custom settings from the old config to the new one (plot directories, pool information, database path, trusted nodes etc) - there are more automated ways to migrate that info if not comfortable editing the config file, yaml is very picky about formatting.
Follow the setup for farming on multiple machines here https://docs.chia.net/farming-on-many-machines/
Start chia and let it sync

Let us know if you run into any issues or have questions during the process.

**Reply by BrandtH22** [reply]
Sounds like a plan, let us know if any issues or questions arise 👍

**Comment by LTMI** [comment]
Thank you so much for your help and have a great day .Best regards,On Jan 9, 2024, at 9:29 AM, ClydeWallace22 ***@***.***> wrote:﻿
Sounds like a plan, let us know if any issues or questions arise 👍

**Reply by BrandtH22** [reply]
Thank you, you as well! 
I am going to close this discussion but if other related questions arise feel free to reopen this discussion, a new one, or join us in our discord :+1:

**Comment by LTMI** [comment]
Hi support,Hope you are well. I have tried to upgrade my chia farmer with 2 harvesters from V1.5 to V2.14 from your recommendation but not successful. I have attached the error messages with this email and jope  you can assists me. Thank you again for your help.Looking to hear from you. C:\Users\Chia_Farmer_HP_SSD1\AppData\Local\Programs\Chia>chia.exe initC:\Users\Chia_Farmer_HP_SSD1\AppData\Local\Programs\Chia>Checking if userData migration is neededC:\Users\Chia_Farmer_HP_SSD1\AppData\Roaming\Chia Blockchain\Local Storage\leveldb exists: trueC:\Users\Chia_Farmer_HP_SSD1\.chia\mainnet\gui\Chia Blockchain\Local Storage\leveldb exists: trueC:\Users\Chia_Farmer_HP_SSD1\AppData\Roaming\Chia Blockchain\Local Storage\leveldb\migrated exists: trueMigration needed: falseSetting user data directory to C:\Users\Chia_Farmer_HP_SSD1\.chia\mainnet\gui\Chia BlockchainRunning python executable:child process successNo certChia directory C:\Users\Chia_Farmer_HP_SSD1\.chia\mainnetC:\Users\Chia_Farmer_HP_SSD1\.chia\mainnet already exists, no migration action takenNo certHave certstderr: Traceback (most recent call last):  File "chia\server\start_full_node.py", line 104, in <module>  File "chia\server\start_full_node.py", line 100, in main  File "chia\server\start_service.py", line 321, in async_run  File "asyncio\runners.py", line 44, in run  File "asyncio\base_events.py", line 649, in run_until_complete  File "chia\server\start_full_node.py", line 83, in async_main  File "chia\server\start_service.py", line 195, in run  File "contextlib.py", line 199, in __aenter__  File "chia\server\start_service.py", line 212, in manage  File "contextlib.py", line 619, in enter_async_context  File "contextlib.py", line 199, in __aenter__  File "chia\full_node\full_node.py", line 259, in manage  File "chia\full_node\block_store.py", line 49, in createRuntimeError: BlockStore does not support database schema v1[7644] Failed to execute script 'start_full_node' due tostderr:  unhandled exception!From: CK Tan ***@***.***>Date: January 9, 2024 at 11:47:48 AM ESTTo: Chia-Network/chia-blockchain ***@***.***>Cc: Chia-Network/chia-blockchain ***@***.***>, Mention ***@***.***>Subject: Re: [Chia-Network/chia-blockchain] Cannot connect  three harvester to the main farmer node (Discussion #17250)﻿Thank you so much for your quick reply. I will go ahead upgrade to the latest version like you recommended. On Jan 9, 2024, at 10:59 AM, ClydeWallace22 ***@***.***> wrote:﻿
Hey @LTMI  , as a heads up we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia).
What I would do is upgrade all machines to the latest and have new configs created:


upgrade to the latest release candidate (there are peer related optimizations): https://github.com/Chia-Network/chia-blockchain/releases/tag/2.1.4-rc2


While installing that I would remove the current peers.dat file in ~.chia\mainnet\db\peers.dat (this will force the client to connect to new peers upon restarting)


Once the latest RC is installed, have a new config file created:



Stop chia
Rename the config file at .chia\mainnet\config\config.yaml
Run chia init
Copy any custom settings from the old config to the new one (plot directories, pool information, database path, trusted nodes etc) - there are more automated ways to migrate that info if not comfortable editing the config file, yaml is very picky about formatting.
Follow the setup for farming on multiple machines here https://docs.chia.net/farming-on-many-machines/
Start chia and let it sync

Let us know if you run into any issues or have questions during the process.

**Comment by LTMI** [comment]
Hi support,Hope you are well. I have tried to upgrade my chia farmer with 2 harvesters from V1.5 to V2.14 from your recommendation but not successful. I have attached the error messages with this email and jope  you can assists me. Thank you again for your help.Looking to hear from you. C:\Users\Chia_Farmer_HP_SSD1\AppData\Local\Programs\Chia>chia.exe initC:\Users\Chia_Farmer_HP_SSD1\AppData\Local\Programs\Chia>Checking if userData migration is neededC:\Users\Chia_Farmer_HP_SSD1\AppData\Roaming\Chia Blockchain\Local Storage\leveldb exists: trueC:\Users\Chia_Farmer_HP_SSD1\.chia\mainnet\gui\Chia Blockchain\Local Storage\leveldb exists: trueC:\Users\Chia_Farmer_HP_SSD1\AppData\Roaming\Chia Blockchain\Local Storage\leveldb\migrated exists: trueMigration needed: falseSetting user data directory to C:\Users\Chia_Farmer_HP_SSD1\.chia\mainnet\gui\Chia BlockchainRunning python executable:child process successNo certChia directory C:\Users\Chia_Farmer_HP_SSD1\.chia\mainnetC:\Users\Chia_Farmer_HP_SSD1\.chia\mainnet already exists, no migration action takenNo certHave certstderr: Traceback (most recent call last):  File "chia\server\start_full_node.py", line 104, in <module>  File "chia\server\start_full_node.py", line 100, in main  File "chia\server\start_service.py", line 321, in async_run  File "asyncio\runners.py", line 44, in run  File "asyncio\base_events.py", line 649, in run_until_complete  File "chia\server\start_full_node.py", line 83, in async_main  File "chia\server\start_service.py", line 195, in run  File "contextlib.py", line 199, in __aenter__  File "chia\server\start_service.py", line 212, in manage  File "contextlib.py", line 619, in enter_async_context  File "contextlib.py", line 199, in __aenter__  File "chia\full_node\full_node.py", line 259, in manage  File "chia\full_node\block_store.py", line 49, in createRuntimeError: BlockStore does not support database schema v1[7644] Failed to execute script 'start_full_node' due tostderr:  unhandled exception!From: CK Tan ***@***.***>Date: January 9, 2024 at 11:47:48 AM ESTTo: Chia-Network/chia-blockchain ***@***.***>Cc: Chia-Network/chia-blockchain ***@***.***>, Mention ***@***.***>Subject: Re: [Chia-Network/chia-blockchain] Cannot connect  three harvester to the main farmer node (Discussion #17250)﻿Thank you so much for your quick reply. I will go ahead upgrade to the latest version like you recommended. On Jan 9, 2024, at 10:59 AM, ClydeWallace22 ***@***.***> wrote:﻿
Hey @LTMI  , as a heads up we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia).
What I would do is upgrade all machines to the latest and have new configs created:


upgrade to the latest release candidate (there are peer related optimizations): https://github.com/Chia-Network/chia-blockchain/releases/tag/2.1.4-rc2


While installing that I would remove the current peers.dat file in ~.chia\mainnet\db\peers.dat (this will force the client to connect to new peers upon restarting)


Once the latest RC is installed, have a new config file created:



Stop chia
Rename the config file at .chia\mainnet\config\config.yaml
Run chia init
Copy any custom settings from the old config to the new one (plot directories, pool information, database path, trusted nodes etc) - there are more automated ways to migrate that info if not comfortable editing the config file, yaml is very picky about formatting.
Follow the setup for farming on multiple machines here https://docs.chia.net/farming-on-many-machines/
Start chia and let it sync

Let us know if you run into any issues or have questions during the process.
